src/en/darkscience/src/eu/kanade/tachiyomi/extension/en/darkscience/DarkScience.kt

```diff
@@ -0,0 +1,156 @@
+package eu.kanade.tachiyomi.extension.en.darkscience
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.model.UpdateStrategy
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import rx.Observable
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class DarkScience : HttpSource() {
+    private val chapterDateRegex = """/(\d\d\d\d/\d\d/\d\d)/""".toRegex()
+    private val chapterNumberRegex = """Dark Science #(\d+)""".toRegex()
+
+    override val name = "Dark Science"
+    override val baseUrl = "https://dresdencodak.com"
+    override val lang = "en"
+    override val supportsLatest = false
+
+    private val archiveUrl = "$baseUrl/category/darkscience/"
+    private val thumbnailUrl = "https://dresdencodak.com/wp-content/uploads/2019/03/DC_CastIcon_Kimiko.png"
+    private val patreonUrl = "https://www.patreon.com/dresdencodak"
+    private val seriesTitle = name
+    private val authorName = "Sen (A. Senna Diaz)"
+    private val seriesGenre = "Science Fiction, Mystery, LGBT+"
+    private val seriesDescription = "" +
+        "Scientist Kimiko Ross has a problem: " +
+        "her money’s gone and a bank exploded her house. " +
+        "With no place else to go, she travels to " +
+        "Nephilopolis, the city of giants – built from the " +
+        "ruins of an ancient war and a fading memory of " +
+        "tomorrow.\n" +
+        "Follow our cyborg hero as she attempts to survive " +
+        "the bureaucratic behemoth with a little “help” " +
+        "from her “friends.” And what exactly is " +
+        "Dark Science anyway?\n" +
+        "Support the comic on Patreon: $patreonUrl"
```

Comment:
you can inline most of these

Reply:
True. I took inspiration from another extension there, which also kept this metadata out of the way from the »do stuff« code, and liked that. I’ll inline if you prefer that, though.

Reply:
All properties that were used only once have been moved now, with the exception of the long description.